### PR TITLE
Redhat ubi

### PR DESCRIPTION
**Pull request by xli1996**
cp-base built based on ubi8-minimal, the size of image will be 677MB

jira : https://confluentinc.atlassian.net/browse/ST-2462

### CODE REVIEW

File: base/Dockerfile [hunk 1]
```diff
@@ -37,15 +37,28 @@ ENV LANG="C.UTF-8"
 # Set the classpath for JARs required by `cub`
 ENV CUB_CLASSPATH='"/usr/share/java/cp-base-new/*"'
 
+# Zulu openJDK
+ENV ZULU_OPENJDK="zulu8.40.0.25-ca-jdk8.0.222-linux_x64"
```
Comment (elismaga) [inline base/Dockerfile]: i don't have a strong feeling about this, but maybe this should go in the common pom

File: base/Dockerfile [hunk 2]
```diff
@@ -37,15 +37,28 @@ ENV LANG="C.UTF-8"
...
+    && mkdir -p /usr/lib/jvm \
+    && cd /usr/lib/jvm \
+    && curl -O https://cdn.azul.com/zulu/bin/${ZULU_OPENJDK}.tar.gz \
+    && tar -xzvf ${ZULU_OPENJDK}.tar.gz \
+    && rm -rf ${ZULU_OPENJDK}.tar.gz \
+    && cd / \
+    && ln -s /usr/lib/jvm/${ZULU_OPENJDK}/bin/java /usr/bin/java \
+    && export JAVA_HOME=/usr/lib/jvm/${ZULU_OPENJDK} \
+    && export PATH=$JAVA_HOME/bin:$PATH \
+    && yum remove -y git \
```
Comment (elismaga) [inline base/Dockerfile]: do you know why we remove pip on the current base image? not saying we should be doing on that on this new image, just kind of wondering why.

**Comment by mohnishbasha** [comment]
This seems like a major change. 
it will be nice if we can we add some additional as to why we are doing this, if a jira is available refer that in the pr and also call out how this was tested.

**Comment by xli1996** [comment]
> This seems like a major change.
> it will be nice if we can we add some additional as to why we are doing this, if a jira is available refer that in the pr and also call out how this was tested.

Since we need to support operator to have redhat version of cp images, and debian base images does not update CVE very frequently, we decide to switch base image to redhat ubi image, here is the one pager https://confluentinc.atlassian.net/wiki/spaces/PM/pages/937297702/1-Pager+Enhance+Production+Quality+of+Confluent+CP+Base+and+Operator+Container+Images . I think the QE team will handle the test part.

File: base/Dockerfile (hunk 2 above)
Comment (xli1996) [reply base/Dockerfile]: since pip is part of python, I do not think we supposed to remove python.

File: base/Dockerfile (hunk 1 above)
Comment (xli1996) [reply base/Dockerfile]: Do you think it should go to common-docker's pom or common's pom?

File: base/Dockerfile [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM azul/zulu-openjdk-debian:8
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
```
Comment (mohnishbasha) [inline base/Dockerfile]: Is it by design that we want to pin this to latest and would we run into any issues down the line due to this?

Comment (elismaga) [reply base/Dockerfile]: For new releases we do want to always get the latest version so that we get all the security updates.

**Pull request merged**